src/RedisDiscover.cpp

```diff
@@ -0,0 +1,83 @@
+#include "RedisDiscover.h"
+ 
+RedisDiscover::~RedisDiscover() {
+    redisFree(redis_client_);
+}
+
+std::string RedisDiscover::getRedisHostNameFromEnv() {
+    // TODO: read URL/Hostname from env variable
+    return "redis";
+}
+
+void RedisDiscover::connect() {
+    struct timeval timeout = { 1, 500000 };
+    int port = 6379;  // default Redis port
+    std::string hostname = getRedisHostNameFromEnv();
+
+    redis_client_ = redisConnectWithTimeout(hostname.c_str(), port, timeout);
+    if (redis_client_ == NULL || redis_client_->err) {
+        if (redis_client_) {
+            char err [128];
+            strcpy(err, redis_client_->errstr);
```

Comment:
are you sure 128 is enough? std::string s = redis_client_->errstr would make a copy for any length